client/src/index.jsx

```diff
@@ -86,11 +93,75 @@ class Zillow extends React.Component {
         main: [],
         group: []
       },
+      value: '',
       mainPage: true,
       index: {}
     };
+    this.deletePhoto = this.deletePhoto.bind(this);
+    this.closePhotoModal = this.closePhotoModal.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+  updatePhoto(e) {
+    var path = window.location.pathname.split('/');
+    var house = path[1];
+    var self = this;
+    var url = this.state.value;
+    $.ajax({
+      method: 'PUT',
+      url: `http://localhost:3002/gallery/${house}/photo/${this.state.photoSelected}/url/${url}`,
+      success: function(data) {
+        //update gallery
+        $.ajax({
+          method: 'GET',
+          url: `http://localhost:3002/gallery/${house}`,
+          success: function(data) {
+            var result = {
+              main: data[0],
+              group: data.slice(1, 5)
+            };
+            self.setState({
+              houseId: house, 
+              gallery: data, 
+              display: result
+            });
+            self.closePhotoModal();
+          } 
+        });
+      }
+    });
+    this.closePhotoModal();
+  }
+  handleChange(event) {
+    this.setState({value: event.target.value});
+  }
+  deletePhoto(e) {
+    var path = window.location.pathname.split('/');
+    var house = path[1];
+    var self = this;
```

Comment:
Hey Bernard, I did something similar, but Leslie on a code review mentioned you can use bind though I think it is harder especially with the mutiple layers, basically you .bind(this) on the outer ajax and on the success and on the inner ajax and on its success.   She says this style is actually an older style of how to do that also think on other request frameworks that are not ajax may be easier than this style.  You may also use arrow functions instead of regular function on other frameworks like axios which just passes the this down.